src/main/java/com/sendgrid/Client.java

```diff
@@ -277,23 +278,19 @@ public Response delete(Request request) throws URISyntaxException, IOException {
 	}
 
 	private Response executeApiCall(HttpRequestBase httpPost) throws IOException {
-		CloseableHttpResponse serverResponse = null;
-		Response response = new Response();
+		CloseableHttpResponse serverResponse = httpClient.execute(httpPost);
```

Comment:
I don't understand how this fixes https://github.com/sendgrid/sendgrid-java/issues/165. It seems like the original code should be closing the connection at the end. Could you please elaborate on how your solution fixes the issue?

Some other questions:
1. Did you test this? If so, could you describe them? I'd like to reproduce.
2. By moving the `httpClient.execute` out of the try/catch/finally, how will we handle any exceptions at that function call.

Thanks!